rpe_pair option

Hello, I've been running pydesigner this way:
    python3.6 pydesigner.py \
    -o $outpath \
    --force --denoise --extent 5,5,5 --degibbs --rician \
    --undistort --smooth \
    --wmti --mask --verbose \
    $dwi1,$dwi2

However, I realize that the default rpe option is rpe_header. It's hardcoded in the script, so I'm wondering if we'll eventually be able to use the rpe_pair option where we can input image of the opposite phase encoding.

Thank you!

Ok. I see. I tried running with my rpe image, but I'm getting this error when pydesigner is converting it to .mif:
[ERROR] bvecs and bvals files must have same number of diffusion directions as DW-image (gradients: 1, image: 81)

it looks like it is trying to run: 
mrconvert -stride 1,2,3,4 -force -fslgrade bvec bval -json_import json dwi2.mif

mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [100%] copying from "/mnt/labsp...JS/M0065/json/DIFF_meso.nii" to "/home/jc83...k/M0065_pydesigner/dwi0.mif"
mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [100%] copying from "/mnt/labsp...n/DIFF_meso_research_s6.nii" to "/home/jc83...k/M0065_pydesigner/dwi1.mif"
mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [ERROR] bvecs and bvals files must have same number of diffusion directions as DW-image (gradients: 1, image: 81)
mrconvert: [ERROR] error importing diffusion gradient table for image "/mnt/labspace/Projects/MESO/ALLSUBJS/M0065/json/DIFF_meso_PA_s7.nii"
Traceback (most recent call last):
  File "/home/jc8329/PyDesigner/pydesigner.py", line 823, in <module>
    main()
  File "/home/jc8329/PyDesigner/pydesigner.py", line 238, in main
    resume=args.resume)
  File "/home/jc8329/PyDesigner/designer/preprocessing/util.py", line 495, in cat
    raise Exception('Please use the "--force" flag to '
Exception: Please use the "--force" flag to overwrite existing outputs, or clear the output directory

Hello, we were able to install miniconda and created an environment to install pydesigner successfully:
pip install -e .
Obtaining file:///cbi05data/data1/Hamster/Projects/PyDesigner
Requirement already satisfied: numpy in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (1.18.2)
Requirement already satisfied: scipy in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (1.4.1)
Requirement already satisfied: matplotlib in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (3.2.1)
Requirement already satisfied: py-cpuinfo in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (5.0.0)
Requirement already satisfied: joblib in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (0.14.1)
Requirement already satisfied: tqdm in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (4.44.1)
Requirement already satisfied: multiprocess in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (0.70.9)
Requirement already satisfied: nibabel in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (3.0.2)
Requirement already satisfied: cvxpy in /usr/local/miniconda3/lib/python3.7/site-packages (from PyDesigner==0.2) (1.0.29)
Requirement already satisfied: python-dateutil>=2.1 in /usr/local/miniconda3/lib/python3.7/site-packages (from matplotlib->PyDesigner==0.2) (2.8.1)
Requirement already satisfied: cycler>=0.10 in /usr/local/miniconda3/lib/python3.7/site-packages (from matplotlib->PyDesigner==0.2) (0.10.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/miniconda3/lib/python3.7/site-packages (from matplotlib->PyDesigner==0.2) (1.2.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/miniconda3/lib/python3.7/site-packages (from matplotlib->PyDesigner==0.2) (2.4.7)
Requirement already satisfied: dill>=0.3.1 in /usr/local/miniconda3/lib/python3.7/site-packages (from multiprocess->PyDesigner==0.2) (0.3.1.1)
Requirement already satisfied: ecos>=2 in /usr/local/miniconda3/lib/python3.7/site-packages (from cvxpy->PyDesigner==0.2) (2.0.7.post1)
Requirement already satisfied: osqp>=0.4.1 in /usr/local/miniconda3/lib/python3.7/site-packages (from cvxpy->PyDesigner==0.2) (0.6.1)
Requirement already satisfied: scs>=1.1.3 in /usr/local/miniconda3/lib/python3.7/site-packages (from cvxpy->PyDesigner==0.2) (2.1.2)
Requirement already satisfied: six>=1.5 in /usr/local/miniconda3/lib/python3.7/site-packages (from python-dateutil>=2.1->matplotlib->PyDesigner==0.2) (1.14.0)
Requirement already satisfied: future in /usr/local/miniconda3/lib/python3.7/site-packages (from osqp>=0.4.1->cvxpy->PyDesigner==0.2) (0.18.2)
Installing collected packages: PyDesigner
  Attempting uninstall: PyDesigner
    Found existing installation: PyDesigner 0.2
    Uninstalling PyDesigner-0.2:
      Successfully uninstalled PyDesigner-0.2
  Running setup.py develop for PyDesigner
Successfully installed PyDesigner

However, I'm still getting this error when I run pydesigner in this conda env:
mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [100%] copying from "/mnt/labsp...JS/M0065/json/DIFF_meso.nii" to "/home/jc83...k/M0065_pydesigner/dwi0.mif"
mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [100%] copying from "/mnt/labsp...n/DIFF_meso_research_s6.nii" to "/home/jc83...k/M0065_pydesigner/dwi1.mif"
mrconvert: [WARNING] existing output files will be overwritten
mrconvert: [ERROR] bvecs and bvals files must have same number of diffusion directions as DW-image (gradients: 1, image: 81)
mrconvert: [ERROR] error importing diffusion gradient table for image "/mnt/labspace/Projects/MESO/ALLSUBJS/M0065/json/DIFF_meso_PA_s7.nii"
Traceback (most recent call last):
  File "/usr/local/miniconda3/bin/pydesigner", line 11, in <module>
    load_entry_point('PyDesigner', 'console_scripts', 'pydesigner')()
  File "/cbi05data/data1/Hamster/Projects/PyDesigner/designer/pydesigner.py", line 238, in main
    resume=args.resume)
  File "/cbi05data/data1/Hamster/Projects/PyDesigner/designer/preprocessing/util.py", line 495, in cat
    raise Exception('Please use the "--force" flag to '
Exception: Please use the "--force" flag to overwrite existing outputs, or clear the output directory

--- a/designer/preprocessing/util.py
+++ b/designer/preprocessing/util.py
@@ -157,7 +157,7 @@ def json2fslgrad(path):
         else:
             nDWI = 1
         if nDWI <= 15:
-            bval = np.zeros(nDWI, dtype=int)
+            bval = np.zeros((1, nDWI), dtype=int)
             bvec = np.zeros((3, nDWI), dtype=int)
             fPath = op.splitext(path)[0]
             np.savetxt(op.join(image.getPath(), image.getName() + '.bvec'), 

